beginner_source/maskedtensor_overview.rst

```diff
@@ -0,0 +1,244 @@
++++++++++
+
+MaskedTensor vs NumPy's MaskedArray
+-----------------------------------
+
+NumPy's ``MaskedArray`` has a few fundamental semantics differences from MaskedTensor.
+
+1. Their factory function and basic definition inverts the mask (similar to ``torch.nn.MHA``); that is, MaskedTensor
+uses ``True`` to denote "specified" and ``False`` to denote "unspecified", or "valid"/"invalid", whereas NumPy does the
+opposite.
+2. Intersection semantics. In NumPy, if one of two elements are masked out, the resulting element will be
+masked out as well -- in practice, they
+`apply the logical_or operator <https://github.com/numpy/numpy/blob/68299575d8595d904aff6f28e12d21bf6428a4ba/numpy/ma/core.py#L1016-L1024>`__.
+
+    >>> data = torch.arange(5.)
+    >>> mask = torch.tensor([True, True, False, True, False])
+    >>> npm0 = np.ma.masked_array(data.numpy(), (~mask).numpy())
+    >>> npm1 = np.ma.masked_array(data.numpy(), (mask).numpy())
+    >>> npm0
+    masked_array(data=[0.0, 1.0, --, 3.0, --],
+                mask=[False, False,  True, False,  True],
+          fill_value=1e+20,
+                dtype=float32)
+    >>> npm1
+    masked_array(data=[--, --, 2.0, --, 4.0],
+                mask=[ True,  True, False,  True, False],
+          fill_value=1e+20,
+                dtype=float32)
+    >>> npm0 + npm1
+    masked_array(data=[--, --, --, --, --],
+                mask=[ True,  True,  True,  True,  True],
+          fill_value=1e+20,
+                dtype=float32)
+
+Meanwhile, MaskedTensor does not support addition or binary operators with masks that don't match -- to understand why,
+please find the section on reductions.
+
+    >>> mt0 = masked_tensor(data, mask)
+    >>> mt1 = masked_tensor(data, ~mask)
+    >>> m0
+    MaskedTensor(
+      [  0.0000,   1.0000,       --,   3.0000,       --]
+    )
+    >>> mt0 = masked_tensor(data, mask)
+    >>> mt1 = masked_tensor(data, ~mask)
+    >>> mt0
+    MaskedTensor(
+      [  0.0000,   1.0000,       --,   3.0000,       --]
+    )
+    >>> mt1
+    MaskedTensor(
+      [      --,       --,   2.0000,       --,   4.0000]
+    )
+    >>> mt0 + mt1
+    ValueError: Input masks must match. If you need support for this, please open an issue on Github.
+
+However, if this behavior is desired, MaskedTensor does support these semantics by giving access to the data and masks
+and conveniently converting a MaskedTensor to a Tensor with masked values filled in using :func:`to_tensor`.
+
+    >>> t0 = mt0.to_tensor(0)
+    >>> t1 = mt1.to_tensor(0)
+    >>> mt2 = masked_tensor(t0 + t1, mt0.get_mask() & mt1.get_mask())
+    >>> t0
+    tensor([0., 1., 0., 3., 0.])
+    >>> t1
+    tensor([0., 0., 2., 0., 4.])
+    >>> mt2
+    MaskedTensor(
+      [      --,       --,       --,       --,       --]
+
+.. _reduction-semantics:
+
+Reduction semantics
+-------------------
+
+The basis for reduction semantics `has been documented and discussed at length <https://github.com/pytorch/rfcs/pull/27>`__,
```

Comment:
I feel like this can be rephrased.

The purpose of a tutorial is to be a self contained, helpful, end to end showcase of a new feature. It is not assumed that the reader has seen RFCs. 

I would start off explaining what a reduction semantic is (we ignore masked values for your favorite functions!), then give the code examples before, then conclude by saying "for more details on the basis for reduction semantics, see this RFC".